spec.html

```diff
@@ -107,6 +107,30 @@ <h1>Well-Known Intrinsic Objects</h1>
 <emu-clause id="sec-abstract-operations">
   <h1>Abstract Operations</h1>
 
+  <emu-clause id="sec-SetterThatIgnoresPrototypeProperties" type="abstract operation">
+    <h1>
+      SetterThatIgnoresPrototypeProperties (
+        _home_: an Object,
+        _p_: a property key,
+        _v_: an ECMAScript language value,
+      ): either a normal completion containing ~unused~ or a throw completion
+    </h1>
+    <dl class="header">
+    </dl>
+    <emu-alg>
+      1. Let _O_ be ? RequireObjectCoercible(*this* value).
+      1. If _O_ is _home_, then
+        1. NOTE: Throwing here emulates assignment to a non-writable data property on the _home_ object in strict mode code.
+        1. Throw a *TypeError* exception.
+      1. Let _desc_ be ? _O_.[[GetOwnProperty]](_p_).
+      1. If _desc_ is *undefined*, then
+        1. Perform ? CreateDataPropertyOrThrow(_O_, _p_, _v_).
+      1. Else,
+        1. Perform ? Set(_O_, _p_, _v_, *true*).
```

Comment:
This will perform `_O_.[[GetOwnProperty]]` a second time, which is pretty weird - are you sure that's what you want? I'm pretty sure that's observable with a Proxy.

What if we just throw in this case instead? You have to either be pulling this off or using `Reflect.set` to hit this branch (right?), and I feel like we should just not support those cases instead of trying to figure out the "right" behavior.

Reply:
The second `[[GetOwnProperty]]` is not ideal, and could theoretically be avoided by extending the MOP, but I don't think we should do that, and I'm not too worried about it.